src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -307,6 +375,27 @@ internal TestExtensions GetExtensionsDiscoveredFromAssembly(string extensionAsse
         }
     }
 
+    /// <summary>
+    /// Gets the cached extensions for the current process.
+    /// </summary>
+    /// 
+    /// <returns>A map representing the cached extensions for the current process.</returns>
+    internal IDictionary<string, ISet<string>> GetCachedExtensions()
+    {
+        var extensionMap = new Dictionary<string, ISet<string>>();
+
+        // Write all "known" cached extension.
```

Comment:
What "known" means here?

Reply:
There's a total of 7 extensions we cache, defined by properties at the top of the file. Pasting them here for convenience:

```
    /// <summary>
    /// Gets or sets test discoverer extensions.
    /// </summary>
    internal Dictionary<string, TestDiscovererPluginInformation> TestDiscoverers { get; set; }

    /// <summary>
    /// Gets or sets test executor extensions.
    /// </summary>
    internal Dictionary<string, TestExecutorPluginInformation> TestExecutors { get; set; }

    /// <summary>
    /// Gets or sets test executor 2 extensions.
    /// </summary>
    internal Dictionary<string, TestExecutorPluginInformation2> TestExecutors2 { get; set; }

    /// <summary>
    /// Gets or sets test setting provider extensions.
    /// </summary>
    internal Dictionary<string, TestSettingsProviderPluginInformation> TestSettingsProviders { get; set; }

    /// <summary>
    /// Gets or sets test logger extensions.
    /// </summary>
    internal Dictionary<string, TestLoggerPluginInformation> TestLoggers { get; set; }

    /// <summary>
    /// Gets or sets test logger extensions.
    /// </summary>
    internal Dictionary<string, TestRuntimePluginInformation> TestHosts { get; set; }

    /// <summary>
    /// Gets or sets data collectors extensions.
    /// </summary>
    internal Dictionary<string, DataCollectorConfig> DataCollectors { get; set; }
```